pytensor/tensor/rewriting/linalg.py

```diff
@@ -310,3 +320,65 @@ def local_log_prod_sqr(fgraph, node):
 
         # TODO: have a reduction like prod and sum that simply
         # returns the sign of the prod multiplication.
+
+
+def local_inv_kron_to_kron_inv(fgraph, node):
+    # check if we have a kron
+    # check if parent node is an inv
+    # if yes, replace with kron(inv, inv)
+
+    pass
+
+
+def local_chol_kron_to_kron_chol(fgraph, node):
+    # check if we have a kron
+    # check if parent node is a cholesky
+    # if yes, replace with kron(cholesky, cholesky)
+
+    pass
+
+
+@register_specialize
+@node_rewriter([Blockwise])
+def local_lift_through_linalg(fgraph, node):
+    """
+    Rewrite a graph like Inv(BlockDiag([A, B, C])) to BlockDiag([Inv(A), Inv(B), Inv(C)])
+
+    Parameters
```

Comment:
Remove boilerplate from docstrings?

Reply:
Expanded out the docstring, curious if you like it or if you prefer them all to be short and sweet

Reply:
I like the explanation a lot. The type hints in the comments not so much, they should be in code?

Definitely prefer this to no docstrings

Reply:
Added them in code, but I had to wrestle with mypy, so the cost is some ugly `cast`ing.

Numpy style guys wants typehints in the docstring, see [here](https://numpydoc.readthedocs.io/en/latest/format.html#parameters). I removed the `typing` style, though. I think it looks neater now.

Reply:
Note they are not vanilla type hints but human readable ones in those examples.

Reply:
Looks good. I am not sure about documenting the signature of functions like this because we have so many, they're internal and always decorated.

Not against it either. Just unsure. Either way the PR was already approved. Merge when you're happy